torchao/float8/inference.py

```diff
@@ -196,6 +198,36 @@ def _is_tensorwise_scaled(x: torch.Tensor) -> bool:
     )
 
 
+def _is_1_128_scaled(x: torch.Tensor) -> bool:
+    """Checks if a quantized tensor is scaled with a block size of 1x128
+    Args:
+        x: quantized tensor (should have `block_size` attribute)
+    """
+    assert hasattr(x, "block_size"), "Expecting input to have `block_size` attribute"
+    b = x.block_size
+    return len(b) >= 2 and math.prod(b[:-1]) == 1 and b[-1] == 128
```

Comment:
is this per block if `len(b) >= 2`? seems like PerGroup? https://github.com/pytorch/ao/blob/1e473ed94caa060b3fbac96657030c44173759e0/torchao/quantization/utils.py#L723

Reply:
as of today:
1. x.shape can have any rank >= 2
2. block_size is statically assigned during config creation, and is not aware of the rank of x

for example, is x.shape is (1, 4, 8) and block_size is (2, 4), we assume that the leading dim is implied and the scale should be of dim (1, 2, 2)

the code ^ translates between (1) and (2)

Reply:
we'll have to revisit this for 3d parameters though, because block_size will be 128x128 and it will need to support 2d and 3d weights